onecodex/models/collection.py

```diff
@@ -362,6 +368,8 @@ def _collate_results(self, metric=None, include_host=None):
         self._cached["results"] = df
```

Comment:
Suggestion: add a new function `_calculate_collated_results` that does all of the above math but doesn't save anything in cached results. Then, call said method in `_collate_results`. Use new method in `_cluster_by_sample` so that you avoid the cache altogether.

I think maybe we discussed this and realized that it wasn't that easy so apologies if this doesn't make sense.